api/google-setup.js

```diff
@@ -1,6 +1,24 @@
 const passport = require('passport');
 const GoogleStrategy = require('passport-google-oauth20').Strategy;
 require('dotenv').config();
+const User = require("./models/user");
+
+// serialize the user.id to save in the cookie session
+// so the browser will remember the user when login
+passport.serializeUser((user, done) => {
+  done(null, user);
+});
+
+// deserialize the cookieUserId to user in the database
+passport.deserializeUser((id, done) => {
+  User.findById(id)
```

Comment:
Here's a console log of the `id` variable.
![Screen Shot 2020-07-17 at 11 13 19 AM](https://user-images.githubusercontent.com/24263366/87818124-f698bb00-c81e-11ea-838a-69f42d2cca8e.png)

The `id` variable here isn't actually the user id because in `serializeUser()` above the whole user object was serialized. So the `id` variable should be whatever we serialized in the `serializeUser()` function, which is the user object. 

Tho it doesn't seem to cause a problem with mongoose `findById()`. So no problem here. Just something to keep in mind.

Reply:
Yup, I've made the deserialize user function to use the user._id now for findById so it should be good now!